doc/source/rllib-models.rst

```diff
@@ -138,51 +142,70 @@ Custom TF models should subclass the common RLlib `model class <https://github.c
-            logits = linear(last_layer, num_outputs, "action",
-                            normc_initializer(0.01))
-            return logits, last_layer
+    class MyKerasRNN(RecurrentTFModelV2):
+        def __init__(self,
+                     obs_space,
+                     action_space,
+                     num_outputs,
+                     model_config,
+                     name,
+                     cell_size=64):
+            super(MyKerasRNN, self).__init__(obs_space, action_space, num_outputs,
+                                             model_config, name)
+            self.cell_size = cell_size
+
+            # Define input layers
+            input_layer = tf.keras.layers.Input(shape=(None, obs_space.shape[0]))
+            state_in_h = tf.keras.layers.Input(shape=(cell_size, ))
+            state_in_c = tf.keras.layers.Input(shape=(cell_size, ))
+            seq_in = tf.keras.layers.Input(shape=())
+
+            # Send to LSTM cell
+            lstm_out, state_h, state_c = tf.keras.layers.LSTM(
+                cell_size, return_sequences=True, return_state=True, name="lstm")(
+                    inputs=input_layer,
+                    mask=tf.sequence_mask(seq_in),
+                    initial_state=[state_in_h, state_in_c])
+            output_layer = tf.keras.layers.Dense(...)(lstm_out)
+
+            # Create the RNN model
+            self.rnn_model = tf.keras.Model(
+                inputs=[input_layer, seq_in, state_in_h, state_in_c],
+                outputs=[output_layer, state_h, state_c])
+            self.register_variables(self.rnn_model.variables)
+            self.rnn_model.summary()
+
+        @override(RecurrentTFModelV2)
+        def forward_rnn(self, inputs, state, seq_lens):
+            model_out, h, c = self.rnn_model([inputs, seq_lens] + state)
+            return model_out, [h, c]
+
+        @override(ModelV2)
+        def get_initial_state(self):
+            return [
+                np.zeros(self.cell_size, np.float32),
+                np.zeros(self.cell_size, np.float32),
+            ]
+
 
 Batch Normalization
 ~~~~~~~~~~~~~~~~~~~
 
 You can use ``tf.layers.batch_normalization(x, training=input_dict["is_training"])`` to add batch norm layers to your custom model: `code example <https://github.com/ray-project/ray/blob/master/python/ray/rllib/examples/batch_norm_model.py>`__. RLlib will automatically run the update ops for the batch norm layers during optimization (see `tf_policy.py <https://github.com/ray-project/ray/blob/master/python/ray/rllib/policy/tf_policy.py>`__ and `multi_gpu_impl.py <https://github.com/ray-project/ray/blob/master/python/ray/rllib/optimizers/multi_gpu_impl.py>`__ for the exact handling of these updates).
 
-Custom Models (PyTorch)
------------------------
+In case RLlib does not properly detect the update ops for your custom model, you can override the ``update_ops()`` method to return the list of ops to run for updates.
+
+PyTorch Models
+--------------
```

Comment:
Can you add an example of PyTorch model in documentation?

Reply:
Fixed